### PR TITLE
asan: enable more asan and ubsan checks.

### PR DESCRIPTION
**Pull request by g-easy**

### CODE REVIEW

File: api/test/nostd/unique_ptr_test.cc [hunk 1]
```diff
@@ -108,8 +108,7 @@ TEST(UniquePtrTest, PointerOperators)
   unique_ptr<int> ptr1{value};
 
   EXPECT_EQ(&*ptr1, value);
-  EXPECT_EQ(
-      unique_ptr<B> {}->f(), 123);
+  EXPECT_EQ(nullptr, (unique_ptr<B> {}).get());
```
Comment (pyohannes) [inline api/test/nostd/unique_ptr_test.cc:111]: I think this test should rather properly initialize a `unique_ptr<B>` and test calling `f`. Like:
```c++
unique_ptr<int> ptr2{new B};
EXPECT_EQ(123, ptr2->f());
```

Comment (g-easy) [reply api/test/nostd/unique_ptr_test.cc]: Done here and also in the shared_ptr test.

**Comment by g-easy** [comment]
PTAL.

Also please clean up the description before merging, or let me merge it. :)

**Comment by g-easy** [comment]
Hmm I was too slow to merge this and now `//sdk/test/metrics:counter_aggregator_test` is failing.

**Pull request merged**